packages/odie-client/src/components/message/feedback-form.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+	useRateChat,
+	useAuthenticateZendeskMessaging,
+	getBadRatingReasons,
+	isTestModeEnvironment,
+} from '@automattic/zendesk-client';
+import { Button, TextareaControl, SelectControl, Spinner } from '@wordpress/components';
+import { useI18n } from '@wordpress/react-i18n';
+import clsx from 'clsx';
+import { useCallback, useEffect, useRef, useState, useMemo } from 'react';
+import Smooch from 'smooch';
+import { useOdieAssistantContext } from '../../context';
+import { useSendChatMessage } from '../../hooks';
+import { Message, MessageAction } from '../../types';
+import { ThumbsDownIcon, ThumbsUpIcon } from './thumbs-icons';
+
+type FeedbackFormProps = {
+	chatFeedbackOptions: MessageAction[];
+};
+
+const generateFeedbackMessage = ( score: 'good' | 'bad' ): Message => {
+	return {
+		content: score === 'good' ? 'Good 👍' : 'Bad 👎',
+		payload: JSON.stringify( { csat_rating: score.toUpperCase() } ),
+		metadata: { rated: true },
+		role: 'user',
+		type: 'message',
+	} as Message;
+};
+
+export const FeedbackForm = ( { chatFeedbackOptions }: FeedbackFormProps ) => {
+	const { isUserEligibleForPaidSupport } = useOdieAssistantContext();
+	const user = Smooch.getUser();
+	const { __ } = useI18n();
+	const [ score, setScore ] = useState< 'good' | 'bad' | '' >( '' );
+	const [ comment, setComment ] = useState( '' );
+	const [ reason, setReason ] = useState( '' );
+	const { data: authData } = useAuthenticateZendeskMessaging(
+		isUserEligibleForPaidSupport,
+		'messenger'
+	);
+	const feedbackRef = useRef< HTMLDivElement | null >( null );
+	const ticketId = useMemo( () => {
+		if ( ! chatFeedbackOptions.length ) {
+			return null;
+		}
+		return chatFeedbackOptions[ 0 ]?.metadata?.ticket_id ?? null;
+	}, [ chatFeedbackOptions ] );
+	const sendMessage = useSendChatMessage();
+
+	const badRatingReasons = getBadRatingReasons();
+
+	const { isPending: isSubmitting, mutateAsync: rateChat } = useRateChat();
+
+	useEffect( () => {
+		if ( score && feedbackRef?.current ) {
+			feedbackRef.current.scrollIntoView( { behavior: 'smooth', block: 'start' } );
+		}
+	}, [ score ] );
+
+	const postCSAT = useCallback( async () => {
+		if ( ! authData?.jwt || ! ticketId || ! score ) {
+			return;
+		}
+
+		await sendMessage( generateFeedbackMessage( score ) );
```

Comment:
Could we send the positive or negative feedback when the user clicks on the thumbs up or down?